include/InverseKinematicsIPOPT.h

```diff
@@ -0,0 +1,101 @@
+#ifndef INVERSEKINEMATICSIPOPT_H
+#define INVERSEKINEMATICSIPOPT_H
+
+#include <IpTNLP.hpp>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/Transform.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/ModelIO/ModelLoader.h>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/KinDynComputations.h>
+#include <math.h>
+#include <IpIpoptApplication.hpp>
+#include <string>
+#include <vector>
+#include <Eigen/Core>
+#include <iDynTree/Core/EigenHelpers.h>
```

Comment:
Every object whose size is needed at compile time (that is, every object allocated on the stack) must be in "included". All references and pointers can be forward declared. 
All not necessary headers  in the `.h` must be included in the `.cpp` file